core/src/main/java/org/infinispan/interceptors/IsMarshallableInterceptor.java

```diff
@@ -95,15 +95,20 @@ public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) t
 
    @Override
    public Object visitLockControlCommand(TxInvocationContext ctx, LockControlCommand command) throws Throwable {
-      if (isStoreAsBinary() || isClusterInvocation(ctx, command))
-         checkMarshallable(command.getKeys());
+      if (isStoreAsBinary() || isClusterInvocation(ctx, command)) {
```

Comment:
Did you mean to include this change in here?  This doesn't seem to relate to this PR.  Maybe another commit for reduce IsMarshallableInterceptor allocations?